flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentInternal.java

```diff
@@ -81,6 +90,15 @@ public interface TableEnvironmentInternal extends TableEnvironment {
      * @param operation The operation to be executed.
      * @return the content of the execution result.
      */
+    TableResultInternal executeOperation(Operation operation);
```

Comment:
This method is quite confusing that what's the difference between this one and `executeInternal`. Besides, why there are no `executeOperation(List<ModifyOperation> operations)` ?  

Couldn't we just implement the delegation in `executeInternal`?

Reply:
We can just  implement the delegation in `executeInternal`. The reason I add `executeOperation` is to make the whole logic clear,  like first try to use external implementation , and then `executeInternal` which is Flink's implementation.
I'm fine to put the delegation in `executeInternal`. 

The reason Why no `executeOperation(List<ModifyOperation> operations)` is I don't want to expose much to user and I think it's no nesscessary. Now, this method is mainly used by `StatementSetOperation`. And if extern dialect wants to custom the implementation for the operation containing multi-insert,  extern dialect can  return a new operation like `HiveStatementSetOperation` and then still delegate it to `executeOperation(Operation operation)`.

Reply:
This sounds we can encapsulate the logic in the internal implementation instead of exposing them on the interface. From the perspective of users, they shouldn't care about the difference between `executeInternal()` and `executeOperation()`. It's very easy to call the wrong method for the users who use the `TableEnvironmentInternl`.